src/timings/TimingsHandler.php

```diff
@@ -332,4 +342,113 @@ public function reset() : void{
 		$this->recordsByParent = [];
 		$this->timingDepth = 0;
 	}
+
+	/**
+	 * Creates a timings report file locally in the server data folder or a custom location.
+	 * Collects timings data and returns a promise that resolves with the timings lines.
+	 *
+	 * @param string|null $filePath Optional custom directory path. If null, uses server data/timings folder
+	 *
+	 * @phpstan-return Promise<list<string>>
+	 */
+	public static function createReportFile(CommandSender $sender, ?string $filePath = null) : Promise{
+		$timingsPromise = self::requestPrintTimings();
+		$timingsPromise->onCompletion(
+			function(array $lines) use ($sender, $filePath) : void{
+				$server = $sender->getServer();
+
+				if($filePath === null){
+					$timingsFolder = Path::join($server->getDataPath(), "timings");
+				}else{
+					$timingsFolder = $filePath;
+				}
+
+				if(!file_exists($timingsFolder)){
+					mkdir($timingsFolder, 0777, true);
+				}
+				$date = date('Y-m-d_H.i.s_T');
+				$index = 0;
+				$timingsFile = Path::join($timingsFolder, "{$date}.txt");
```

Comment:
Probably better to keep some kind of prefix on this so it can be visually distinguished from a crashdump